addons/account/models/account_journal_dashboard.py

```diff
@@ -694,6 +698,34 @@ def _get_journal_dashboard_bank_running_balance(self):
             )
         return result
 
+    def _get_outstanding_payments_balances(self):
+        self.env.cr.execute("""
+            SELECT move.journal_id AS journal_id,
+                   move.company_id AS company_id,
+                   move.currency_id AS currency,
+                   SUM(CASE
+                       WHEN payment.payment_type = 'outbound' THEN -payment.amount
+                       ELSE payment.amount
+                   END) AS amount_total,
+                   SUM(amount_company_currency_signed) AS amount_total_company
+              FROM account_payment payment
+              JOIN account_move move ON move.payment_id = payment.id
+              JOIN account_journal journal ON move.journal_id = journal.id
+             WHERE payment.is_matched IS TRUE
+               AND move.state = 'posted'
+               AND move.journal_id = ANY(%s)
+               AND move.company_id = ANY(%s)
+               AND payment.outstanding_account_id = journal.default_account_id
+          GROUP BY move.company_id, move.journal_id, move.currency_id
+        """, [self.ids, self.env.companies.ids])
+        query_result = group_by_journal(self.env.cr.dictfetchall())
+        result = {}
+        for journal in self:
+            # User may have read access on the journal but not on the company
+            currency = journal.currency_id or self.env['res.currency'].browse(journal.company_id.sudo().currency_id.id)
```

Comment:
You can use `with_env`to avoid browsing and losing the prefetch (generally speaking, shouldn't have too big of an impact here). It's also shorter :wink: 
```suggestion
            currency = (journal or journal.company_id.sudo()).currency_id.with_env(self.env)
```

Reply:
Actually, it does not work, it returns an empty recordset

Reply:
Then
```suggestion
            currency = (journal.currency_id or journal.company_id.sudo().currency_id).with_env(self.env)
```